alt_src/alt_src.py

```diff
@@ -263,20 +265,24 @@ def get_output(self, cmd, fatal=True, **kwargs):
         if 'stderr' in kwargs:
             # convenience values
             if kwargs['stderr'] == 'null':
-                kwargs['stderr'] = file('/dev/null', 'w')
+                kwargs['stderr'] = open('/dev/null', 'w')
             elif kwargs['stderr'] == 'keep':
                 kwargs['stderr'] = subprocess.STDOUT
         elif self.logfile:
             self.logfile.flush()
             kwargs['stderr'] = self.logfile
         proc = subprocess.Popen(cmd, **kwargs)
         output = proc.communicate()[0]
+        # we'll decode it only if it's bytes type (not a PY2 'str' type)
```

Comment:
Instead of writing our own code to do this, can we please use the built-in features of Popen instead?

I believe what you want here is to pass the oddly named "universal_newlines=True" option which will cause the output to be read in text mode on Python 3, and is also compatible with Python 2.  It'll result in the same return types as you currently have with this explicitly written code (i.e. 'str' on both Python 2 and Python 3).

Reply:
Fixed.